modules/strelka/1.0/strelka.smk

```diff
@@ -0,0 +1,256 @@
+
+
+# Original Author:  Lauren Chong
+# Module Author:    Helena Winata
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["strelka"]`
+CFG = op.setup_module(
+    name = "strelka",
+    version = "1.0",
+    subdirectories = ["inputs", "chrom_bed", "strelka", "filtered", "outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _strelka_input_bam,
+    _strelka_input_vcf,
+    _strelka_dummy_vcf,
+    _strelka_index_bed,
+    _strelka_configure_paired,
+    _strelka_configure_unpaired,
+    _strelka_decompress_vcf,
+    _strelka_filter,
+    _strelka_output_vcf,
+    _strelka_all,
+
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _strelka_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+
+rule _strelka_dummy_vcf:
+    output:
+        touch(CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/vcf/{tumour_id}--{normal_id}--{pair_status}.dummy.tbi")
+
+
+#if not CFG["inputs"]["candidate_small_indels_tbi"] and CFG["inputs"]["candidate_small_indels_vcf"]:
+rule _strelka_input_vcf:
+    input:
+        dummy_vcf = CFG["inputs"]["candidate_small_indels_output"]
+    params:
+        vcf = CFG["inputs"].get("candidate_small_indels_vcf") or "",
+        tbi = CFG["inputs"].get("candidate_small_indels_tbi") or "",
+        
+    output:
+        vcf = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/vcf/{tumour_id}--{normal_id}--{pair_status}.candidateSmallIndels.vcf.gz",
+        tbi = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/vcf/{tumour_id}--{normal_id}--{pair_status}.candidateSmallIndels.vcf.gz.tbi"
+    run:
+        shell("touch {params.in_tbi}")
+        op.relative_symlink(params.vcf, output.vcf)
+        op.relative_symlink(params.tbi, output.tbi)
+
+
+
+# bgzip-compress and tabix-index the BED file to meet strelka requirement
+rule _strelka_index_bed:
+    input:
+        bed = reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.bed")
+    output:
+        bedz = CFG["dirs"]["chrom_bed"] + "{genome_build}.main_chroms.bed.gz"
+    conda:
+        CFG["conda_envs"]["tabix"]
+    shell:
+        op.as_one_line("""
+        bgzip -c {input.bed} > {output.bedz}
+            &&
+        tabix {output.bedz}
+        """)
+
+def _get_indel_cli_arg(vcf_in = config["lcr-modules"]["strelka"]["inputs"]["candidate_small_indels_tbi"]):
```

Comment:
Since there are a few small edits needed before merging, would you prefix all module functions with the module name? Here, it would be `_strelka_get_indel_cli_arg`.

Reply:
okay!